bin/migrate-oats-data/submissions/submap/soil_elements.py

```diff
@@ -0,0 +1,132 @@
+                    soil_dict[app_component_id]['total_remove'] = row[volume]
+                    soil_dict[app_component_id]['remove_duration'] = row[duration]
+                    soil_dict[app_component_id]['remove_area'] = row[area]
+                    if 'import_fill' not in soil_dict.get(app_component_id, {}):
+                        soil_dict[app_component_id]['import_fill'] = False                    
+
+                soil_dict[app_component_id][SoilAction.RMV.name] = True
+
+
+            elif row[code] == SoilAction.ADD.value:
+                if 'ADD' in soil_dict.get(app_component_id, {}):
+                        print('ignored element_id:',row['soil_change_element_id'])
+                else:            
+                    soil_dict[app_component_id]['fill_type'] = row[desc]
+                    soil_dict[app_component_id]['fill_origin'] = row[origin_desc]
+                    soil_dict[app_component_id]['total_fill'] = row[volume]
+                    soil_dict[app_component_id]['max_fill_depth'] = row[depth]
+                    soil_dict[app_component_id]['fill_duration'] = row[duration]
+                    soil_dict[app_component_id]['fill_area'] = row[area]
+                    soil_dict[app_component_id]['import_fill'] = True
+
+                soil_dict[app_component_id][SoilAction.ADD.name] = True
+            
+            else:
+                print('unknown soil action')
+        else:
+            soil_dict[app_component_id] = {}
+            soil_dict[app_component_id][alr_id] = row[alr_id]
+            if row[code] == SoilAction.RMV.value:
+                soil_dict[app_component_id]['remove_type'] = row[desc]
+                soil_dict[app_component_id]['remove_origin'] = row[origin_desc]
+                soil_dict[app_component_id]['max_remove_depth'] = row[depth]
+                soil_dict[app_component_id]['total_remove'] = row[volume]
+                soil_dict[app_component_id]['remove_duration'] = row[duration]
+                soil_dict[app_component_id]['remove_area'] = row[area]
+                soil_dict[app_component_id][SoilAction.RMV.name] = True
+                soil_dict[app_component_id]['import_fill'] = False
+
+
+            elif row[code] == SoilAction.ADD.value:          
+                soil_dict[app_component_id]['fill_type'] = row[desc]
+                soil_dict[app_component_id]['fill_origin'] = row[origin_desc]
+                soil_dict[app_component_id]['total_fill'] = row[volume]
+                soil_dict[app_component_id]['max_fill_depth'] = row[depth]
+                soil_dict[app_component_id]['fill_duration'] = row[duration]
+                soil_dict[app_component_id]['fill_area'] = row[area]
+                soil_dict[app_component_id]['import_fill'] = True
+                soil_dict[app_component_id][SoilAction.ADD.name] = True
+                
+            else:
+                print('unknown soil action')
+    return soil_dict
```

Comment:
This looks very messy and has a ton of repetitive code. Have you tried to clean it up?